src/react/projects/spark-core-react/src/SprkCard/SprkCard.js

```diff
@@ -0,0 +1,175 @@
+    // Extra classes that on the Icon of the CTA teaser
+    additionalCtaIconClasses: PropTypes.string,
+    // Add classes to media when it's icon
+    additionalMediaIconClasses: PropTypes.string,
+    // Text inside of teaser card
+    bodyText: PropTypes.string,
+    // Analytics configuration
+    ctaAnalytics: PropTypes.string,
+    // Decides where the Call to Action or the media will go
+    href: PropTypes.string,
+    linkType: PropTypes.oneOf(['anchor', 'router-link']),
+    // Option to add icon next to Call to Action
+    ctaIcon: PropTypes.string,
+    // Text of the Call to Action
+    ctaText: PropTypes.string,
+    // Chooses if the Call ot Action is a link or a button
+    ctaType: PropTypes.oneOf(['link', 'button']),
+    // alt attribute for img
+    mediaAlt: PropTypes.string,
+    // Option for image link analytic tracking
+    mediaLinkAnalytics: PropTypes.string,
+    // Source of main image
+    mediaSrc: PropTypes.string,
+    // Decides where the main media link will go
+    mediaIconName: PropTypes.string,
+    // REQUIRED
+    // Decides if main media of teaser card is an image or an icon
+    mediaType: PropTypes.oneOf(['img', 'icon']),
+    // Teaser's headline
+    title: PropTypes.string,
+    // Will change the order of teaser from title first to main media first
+    titleFirst: PropTypes.bool,
+  }),
+  teaserConfig: PropTypes.shape({
+    // Text int the main body
+    bodyText: PropTypes.string,
+    cta: PropTypes.shape({
+      // Extra classes that on the Call to Action of Teaser
+      additionalCtaIconClasses: PropTypes.string,
+      // CTA Analytics
+      ctaAnalytics: PropTypes.string,
+      // Icon next to CTA
+      ctaIcon: PropTypes.string,
+      // Anchor tag, or router link at user discrection
+      mediaLinkElement: PropTypes.oneOfType([PropTypes.string, PropTypes.func]),
+      // Link or Button
+      ctaVariant: PropTypes.oneOf(['link', 'button']),
+      // Where the cta goes
+      href: PropTypes.string,
+    }),
+    // Configs for the main media of the teaser card
+    media: PropTypes.shape({
+      // Additional classes if it's an icon
+      additionalMediaIconClasses: PropTypes.string,
+      // Where media will go
+      href: PropTypes.string,
+      //// Anchor tag, or router link at user discrection
+      mediaLinkElement: PropTypes.oneOfType([PropTypes.string, PropTypes.func]),
+      // Reference name of SprkIcon
+      iconName: PropTypes.string,
+      // Alternative text for image
+      imgAlt: PropTypes.string,
+      // src for image
+      imgSrc: PropTypes.string,
+      // Analytics id tag for image
+      mediaAnalyticsString: PropTypes.string,
+      // Chooses if main media is img or SprkIcon
+      mediaVariant: PropTypes.oneOf(['img', 'icon']),
+    }),
+    title: PropTypes.string,
+    titleFirst: PropTypes.bool,
+  }),
+};
+
+SprkCard.defaultProps = {
+  additionalClasses: '',
+  highlightedHeaderConfig: {},
+  children: null,
+  idString: '',
```

Comment:
an empty string here will render `data-id` even if there is no value. `null` would be better, so `data-id` doesn't render unless it has a value.